esphome/components/logger/__init__.py

```diff
@@ -180,6 +181,19 @@ def validate_local_no_higher_than_global(value):
             cv.Optional(CONF_BAUD_RATE, default=115200): cv.positive_int,
             cv.Optional(CONF_TX_BUFFER_SIZE, default=512): cv.validate_bytes,
             cv.Optional(CONF_DEASSERT_RTS_DTR, default=False): cv.boolean,
+            cv.SplitDefault(
+                CONF_TASK_LOG_BUFFER_SIZE,
+                esp32=768,  # Default: 768 bytes (~5-6 messages with 70-byte text plus thread names)
+            ): cv.All(
+                cv.only_on_esp32,
+                cv.Any(
```

Comment:
This will also require the test files to be updated with units.
```suggestion
                cv.validate_bytes,
                cv.Any(
```